crates/bevy_ecs/src/reflect/map_entities.rs

```diff
@@ -1,76 +1,118 @@
 use crate::{
     component::Component,
-    entity::{Entity, EntityHashMap, MapEntities, SceneEntityMapper},
+    entity::{Entity, EntityHashMap, EntityMapper, MapEntities, SceneEntityMapper},
     world::World,
 };
-use bevy_reflect::FromType;
+use bevy_reflect::{FromReflect, FromType, PartialReflect};
 
 /// For a specific type of component, this maps any fields with values of type [`Entity`] to a new world.
 ///
 /// Since a given `Entity` ID is only valid for the world it came from, when performing deserialization
 /// any stored IDs need to be re-allocated in the destination world.
 ///
 /// See [`SceneEntityMapper`] and [`MapEntities`] for more information.
+///
+/// [`MapEntities`]: crate::entity::MapEntities
 #[derive(Clone)]
 pub struct ReflectMapEntities {
-    map_all_entities: fn(&mut World, &mut SceneEntityMapper),
-    map_entities: fn(&mut World, &mut SceneEntityMapper, &[Entity]),
+    map_all_world_entities: fn(&mut World, &mut SceneEntityMapper),
+    map_world_entities: fn(&mut World, &mut SceneEntityMapper, &[Entity]),
```

Comment:
Maybe just remove these two methods instead of deprecation?

Reply:
They're required in the scene writers for now until we reorder how it works. I've got another change like #15422 that builds on this PR instead that I could include, but I thought it best to talk about the `MapEntities` changes in isolation first.